docs/en/observability/logs-troubleshooting.asciidoc

```diff
@@ -1,4 +1,111 @@
 [[logs-troubleshooting]]
 = Troubleshoot your logs
 
-_coming soon_
\ No newline at end of file
+This section provides solutions to errors you might encounter while onboarding your logs in {kib}.
+
+[discrete]
+[[logs-troubleshooting-insufficient-priv]]
+== User does not have permissions to create API key
+
+If you don't have the required privileges to create an API key, you'll see the following error message:
+
+----
+User does not have permissions to create API key.
+
+Required cluster privileges are [`monitor`, `manage_own_api_key`] and 
+required index privileges are [`auto_configure`, `create_doc`] for 
+indices [`logs-*-*`, `metrics-*-*`], please add all required privileges 
+to the role of the authenticated user.
+----
+
+*Solution:*
+
+You need an administrator to give you the necessary permissions listed in the error messages and then to restart the onboarding flow.
+
+[discrete]
+[[logs-troubleshooting-API-key-failed]]
+== Failed to create API key
+
+If you don't have the privileges to create `savedObjects` in {kib}, you'll see the following error message:
+
+----
+Failed to create API key
+
+Something went wrong: Unable to create observability-onboarding-state
+----
+
+*Solution:*
+
+You need an administrator to give you privileges to create `savedObjects` in {kib} to generate the required `observabilityOnboarding` flow state.
+Once you have necessary privileges, restart the onboarding flow.
+
+[discrete]
+[[logs-troubleshooting-kib-not-accessible]]
+== {kib} not accessible from host
+
+If {kib} is not accessible from the host, you'll see the following error message after pasting the *Install the {agent}* instructions your host:
+
+----
+New integrated terminal
+
+To set your external terminal as the default, change your preferences.
+----
+
+*Solution:*
+The host needs access to {Kib}. If that's not possible, you can manually download and configure the {agent}. See the <<logs-stream, Stream a log file>> documentation for more information.
+
+[discrete]
+[[logs-troubleshooting-download-agent]]
+== Download {agent} failed
+
+If the host was able to download the installation script but cannot connect to the public artifactory repository, you'll see the following error message:
+
+----
+Download Elastic Agent
+
+Failed to download Elastic Agent, see script for error.
+----
+
+*Solutions:*
+
+* *Agent version and OS architecture is not available*
++
+The combination of {agent} version and OS architecture is not available. If this is the case you'll see the following error message on your host:
++
+----
+New integrated terminal
+
+To set your external terminal as the default, change your preferences.
+----
++
+To fix this, modify the {agent} version in the installation instructions to a known version of the {agent}.
+
+* You're an Elastic Cloud Enterprise user without access to the Elastic downloads page.
+* The {agent} was fully downloaded previously. In this case, delete previous downloads and restart the onboarding.
```

Comment:
Here the error in the console should be more align to
```
The requested URL returned error: 404
```

Reply:
Concretely this is what we are showing
<img width="888" alt="image" src="https://github.com/elastic/observability-docs/assets/1313018/42d93130-0c38-4112-9b86-327a6433e26a">

What do you suggest @ruflin ?

Reply:
If the agent was already downloaded also the above error shows? I was referring to:

```
The {agent} was fully downloaded previously. In this case, delete previous downloads and restart the onboarding.
```

I'm surprised we see a 404 in this case.

Reply:
ah no, the 404 it's in case the url for the agent returns the 404 or we cannot connect to kibana to download the script.

In the case of an agent already existing in the host, we would probably would see something like
```
Error: already installed at: /opt/Elastic/Agent
```

Let me double check this scenario